R/compare.r

```diff
@@ -145,7 +145,7 @@ compare.numeric <- function(x, y, ..., max_diffs = 10) {
 
   # If they're not the same type or length, fallback to default method
   equal <- paste0(equal, collapse = "\n")
-  if (!is.integer(x) && !is.numeric(y)) return(comparison(FALSE, equal))
```

Comment:
I think that's probably reasonable - it doesn't really matter what the values are if the classes are different.  But a bigger problem is `expect_equal(1, 1L)` which for convenience shouldn't fail. Maybe we should special case integer vs. numeric comparison?

This current behaviour is also bad:

```R
expect_equal(1L, factor(1))
#  Error: 1L not equal to factor(1)
# 1 - 1 == NA In addition: Warning message:
# In Ops.factor(x, y) : ‘-’ not meaningful for factors
```